### ISSUE TITLE
linter: vitest/max-nested-describe don't reset with nested sibilings

### ISSUE THREAD

**Issue by Mister-Hope** [oxc-project/oxc#21888]
### What version of Oxlint are you using?

latest
### What happened?

linter: vitest/max-nested-describe don't reset with nested sibilings.

Basically, https://oxc.rs/docs/guide/usage/linter/rules/vitest/max-nested-describe.html gives a correctness result:

```ts
describe("foo", () => {
  describe("bar", () => {
    it("should get something", () => {
      expect(getSomething()).toBe("Something");
    });
  });
  describe("qux", () => {
    it("should get something", () => {
      expect(getSomething()).toBe("Something");
    });
  });
});

describe("foo2", function () {
  it("should get something", () => {
    expect(getSomething()).toBe("Something");
  });
});

describe("foo", function () {
  describe("bar", function () {
    describe("baz", function () {
      describe("qux", function () {
        describe("this is the limit", function () {
          it("should get something", () => {
            expect(getSomething()).toBe("Something");
          });
        });
      });
    });
  });
});
```

While the result itselfs is now failing:

<img width="1324" height="669" alt="Image" src="https://github.com/user-attachments/assets/0fb2d052-9dd6-41bd-9396-fe6e9256b094" />

This is because though first `foo` ends, but the rule still regard this as a parent for foo2, and foo2 the parent of the second foo.

For me I am maintaining a markdown-it repo containing 1525 test cases, and it produces tons of noise after upgrading.
